pysrc/main.py

```diff
@@ -1,10 +1,50 @@
 """Main Entry point for app_hash sync"""
 import asyncio
+import logging.config
 import os
 
 from dotenv import load_dotenv
+from dune_client.types import DuneRecord
 
 from pysrc.fetch.dune import DuneFetcher
+from pysrc.fetch.ipfs import Cid
+
+log = logging.getLogger(__name__)
+logging.basicConfig(format="%(asctime)s %(levelname)s %(name)s %(message)s")
+log.setLevel(logging.DEBUG)
+
+
+def fetch_and_filter(
+    dune_results: list[DuneRecord],
+    max_retries: int = 3,
+) -> tuple[list[DuneRecord], list[DuneRecord]]:
+    """
+    Run loop fetching app_data for hashes,
+    separates into (found and not found), returning the pair.
+    """
+    found: list[dict[str, str]] = []
+    not_found: list[dict[str, str]] = []
+    # Drain the dune_results into "found" and "not found" categories
+    while dune_results:
+        row = dune_results.pop()
+        app_hash = row["app_hash"]
+        cid = Cid(app_hash)
+        app_data = cid.get_content(max_retries)
```

Comment:
If you don't want to pass the max_retries into the function there's a cool way to do it using decorators.

Ref: https://stackoverflow.com/a/64030200/3841083

Reply:
Cool I will try this out... although I am also proposing (in the description that we eliminate this -- since I don't think it helps).

Reply:
Cool I will try this out... although I am also proposing (in the description that we eliminate this -- since I don't think it helps).

Reply:
> Cool I will try this out... although I am also proposing (in the description that we eliminate this -- since I don't think it helps).

Ah thanks for pointing that out, I missed it. Then I don't think it makes a lot of sense to spend extra time on this. Just wanted to share this cool possibility of doing things with decorators. Maybe it's more useful sometimes in the future.

Reply:
Yes this actually very cool. I think I'll turn this into a "good first issue" so that others have some fun things to do! If we do keep it. I love decorators!

Reply:
I was actually another cool decorator thing today in a different place, unfortunately it was only supported in python >= 3.10 so it will also have to wait:

https://github.com/cowprotocol/dune-client/issues/37#issuecomment-1319901120

Reply:
That looks really nice! Too bad we can't support it in earlier python versions :/